### PR TITLE
openlens: remove package

### PR DESCRIPTION
**Pull request by SebTM**
## Description of changes

Remove package - [see homepage](https://github.com/MuhammedKalkan/OpenLens) - replacement `lens(-desktop)` already packaged.

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [x] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Review by eclairevoyant — COMMENTED** [review]
Please add an entry in `aliases.nix`.

**Comment by SebTM** [comment]
> Please add an entry in `aliases.nix`.

Done

**Comment by eclairevoyant** [comment]
@ofborg eval

File: pkgs/top-level/aliases.nix [hunk 1]
```diff
@@ -1132,6 +1132,7 @@ mapAliases ({
   openjfx19 = throw "OpenJFX 19 was removed as it has reached its end of life"; # Added 2024-08-01
   openjfx20 = throw "OpenJFX 20 was removed as it has reached its end of life"; # Added 2024-08-01
   openjpeg_2 = openjpeg; # Added 2021-01-25
+  openlens = lens; # Added 2024-09-04
```
Comment (eclairevoyant) [inline pkgs/top-level/aliases.nix:1135]: Is it the same app / 100% compatible? I'd rather not have people surprised by the change from openlens to lens

Comment (eclairevoyant) [reply pkgs/top-level/aliases.nix:1135]: Throw sounds safer, yeah

**Review by eclairevoyant — APPROVED** [review]

**Comment by sjdrc** [comment]
Why did this package need to be removed? OpenLens still functions perfectly well in its current state, there's no need to switch to the closed source lens.

**Comment by SebTM** [comment]
For me it was a dead package I could have removed myself as maintainer otherwise - but it does not receive any updates which would worry me security wise at some point - not using it anymore feel free to PR to readd if you think it will work long-term...

**Comment by MartinLoeper** [comment]
I think this is a good decision. It sounds hard to maintain a dead project.

**Pull request merged**